examples/logger/src/bring_your_own_formatter.py

```diff
@@ -1,12 +1,13 @@
 from aws_lambda_powertools import Logger
 from aws_lambda_powertools.logging.formatter import LambdaPowertoolsFormatter
+from aws_lambda_powertools.logging.types import TypedLog
 
 
 class CustomFormatter(LambdaPowertoolsFormatter):
-    def serialize(self, log: dict) -> str:
+    def serialize(self, log: TypedLog) -> str:  # type: ignore
```

Comment:
```suggestion
    def serialize(self, log: PowertoolsLogRecord | Mapping) -> str:  # type: ignore
```

`type: ignore` is a last resort, we should avoid it. What you'll have to do here is to first allow a Typed Dictionary (PowertoolsLogRecord you defined earlier), and any Dictionary to allow inheritance and override (liskov substitution principle).

From here, you'll have to address other parts in the Formatter code which expected a Dict to serialize/deserialize and use Mapping.

Mapping is a generic hashable value, a protocol in Python words, which would satisfy our typing constraints - more info here: https://docs.python.org/3/library/stdtypes.html#mapping-types-dict

Please shout if you get stuck. We should be changing our actual implementation in `formatter.py` too in addition to this example.